tests/integration/test_charm.py

```diff
@@ -352,29 +352,56 @@ async def test_saml_auth(  # pylint: disable=too-many-locals
         assert "Continue to your account" in logged_in_page.text
 
 
+@pytest.mark.parametrize(
+    "relation_name",
+    [
+        pytest.param("smtp-legacy"),
+        pytest.param("smtp", marks=[pytest.mark.requires_secrets]),
+    ],
+)
 async def test_synapse_enable_smtp(
+    model: Model,
     synapse_app: Application,
     get_unit_ips: typing.Callable[[str], typing.Awaitable[tuple[str, ...]]],
     access_token: str,
+    relation_name: str,
 ):
     """
-    arrange: build and deploy the Synapse charm, create an user, get the access token
-        and enable SMTP.
+    arrange: build and deploy the Synapse charm. Create an user and get the access token
+        Deploy, configure and integrate with Synapse the smtp-integrator charm.
     act:  try to check if a given email address is not already associated.
     assert: the Synapse application is active and the error returned is the one expected.
     """
-    await synapse_app.set_config({"smtp_host": "127.0.0.1"})
-    await synapse_app.model.wait_for_idle(
-        idle_period=30, apps=[synapse_app.name], status=ACTIVE_STATUS_NAME
+    if "smtp-integrator" in model.applications:
+        await model.remove_application("smtp-integrator")
+        await model.block_until(lambda: "smtp-integrator" not in model.applications, timeout=60)
+        await model.wait_for_idle(status=ACTIVE_STATUS_NAME, idle_period=5)
+
+    smtp_integrator_app = await model.deploy(
+        "smtp-integrator",
+        channel="latest/edge",
+        config={
+            "auth_type": "plain",
+            "host": "127.0.0.1",
+            "password": "SECRET",
```

Comment:
I'm suprised that bandit doesn't complain, why not just use the `secrets` module to generate something?

Reply:
True. Changed to secrets